docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+# ADR-0008: Digest Calculation & Signing/Pinning (OCI Cosign Support)
+
+- **Status:** Proposed
+- **Deciders:** OCM Maintainers
+- **Date:** 2025-08-27
+- **Related:** OCM signing specification and examples
+
+---
+
+## Context
+
+Component Descriptors (CDs) are signed without altering digest fields during signing. Digests for resources, component references, and the component version are produced in a dedicated step and embedded into the descriptor. Signing uses fixed canonicalization (JCS, RFC 8785) over the descriptor to compute the component-version digest and create a signature envelope. When a component version references another, the digest of the referenced descriptor is embedded in the parent’s `references[]` and contributes to the parent’s component-version digest.
+
+Some workflows require an additional **OCI Cosign** signature over the same canonical payload, optionally uploaded to an OCI registry and recorded in a transparency log (Rekor).
```

Comment:
this is not correct. I dont know where you go the OCI registry upload from but the signature is unrelated to the storage